src/pages/createAppointment/[postId].tsx

```diff
@@ -0,0 +1,118 @@
+  const [availableParticipants, setAvailableParticipants] = useState<User[] | null>(null);
+  const [selectedParticipants, setSelectedParticipants] = useState<User[]>([]);
+  const [participantCountError, setParticipantCountError] = useState<boolean>(false);
+
+  const postId = parseInt(router.query.postId as string);
+
+  function handleClickBack() {
+    router.back();
+  }
+
+  function handleParticipantsChange(participants: User[]): void {
+    setParticipantCountError(false);
+    setSelectedParticipants(participants);
+  }
+
+  function onSubmit() {
+    if (selectedParticipants.length == 0) {
+      setParticipantCountError(true);
+      return;
+    }
+
+    if (postInfo) {
+      CreateAppointment(postId, postInfo, selectedParticipants, supabaseClient)
+        .then(() => {
+          // router.push(viewAppointment);
+          return;
+        })
+        .catch((err) => {
+          console.log(err);
+          return;
+        });
+    }
+  }
+
+  useEffect(() => {
+    if (!postId || !userStatus.user) return;
+    GetPostWithParticipantsByPostId(postId, supabaseClient)
+      .then((p) => {
+        setPostInfo(p);
+        if (!p.participants) return;
+        setAvailableParticipants(p.participants)
+      })
+      .catch((err) => {
+        console.log(err);
+        return;
+      });
+  }, [supabaseClient, postId, userStatus.user]);
+
+  if (userStatus.isLoading) return <Loading />;
+  if (!userStatus.user) {
+    router.push(PagePaths.login);
+    return;
+  }
+  if (!postInfo || !availableParticipants) return <Loading />;
+  return (
+    <>
+      <Navbar />
+      {/*  ArrowBackIcon */}
+      <Box display="flex" paddingBottom="40px">
+        <Link>
+          <ArrowBackIcon
+            fontSize="large"
+            sx={{ position: "absolute", margin: "3vh 0 0 3vh", color: "black" }}
+            onClick={handleClickBack}
+          />
+        </Link>
+      </Box>
+      {/*  Title */}
+      <Box>
+        <Typography variant="h1" sx={{ fontWeight: "700", fontSize: "40px" }}>
+          Create Appointment
+        </Typography>
+      </Box>
+      {/*  Left and Right Card */}
+      <Box display="flex" justifyContent="center" padding="40px">
+        <Grid container spacing="40px" width="80vw">
+          {/*  Left Card */}
+          <Grid item xs={12} md={6} style={{ display: "flex" }}>
+            <LeftCard postInfo={postInfo} isUnClick={true} />
+          </Grid>
+          {/*  Right Card */}
+          <Grid item xs={12} md={6} style={{ display: "flex" }}>
+            <RightCard
+              images={postInfo.images}
+              availableParticipants={availableParticipants}
+              selectedParticipants={selectedParticipants}
+              onSubmit={onSubmit}
+              participantCountError={participantCountError}
+              handleParticipantsChange={handleParticipantsChange}
+            />
+          </Grid>
+        </Grid>
+      </Box>
```

Comment:
I don't think the left and right card should extend the height as the `Create` button appear.

Reply:
งง
ภาษาไทยหน่อย

Reply:
กล่องทั้ง 2 ฝั่งมันไม่ควรยืดตอนมีปุ่ม create เพิ่มขึ้นมา

Reply:
It's a bad design, I think. Should disable or enable the create button instead of hiding it.

Reply:
Changed it to disabling the button instead of hiding.